src/utils/output.ts

```diff
@@ -433,7 +433,7 @@ export class RetrieveResultFormatter implements Formatter<RetrieveResultJson> {
   private async getPackages(): Promise<NamedPackageDir[]> {
     const packages: NamedPackageDir[] = [];
     const projectPath = await SfProject.resolveProjectPath();
-    this.packageNames.forEach((name) => {
+    this.packageNames?.forEach((name) => {
```

Comment:
Is this needed?
I was able to compile and run all the test in this project without it.

Reply:
@RodEsp yes, it fails when running the command against a dx project where the sfdx-project.json does not contain the packages property.